components/encoder/incremental/incremental_encoder.go

```diff
@@ -209,7 +208,7 @@ func (e *Encoder) Start(ctx context.Context, b board.Board, interrupts []string)
 	// x -> impossible state
 
 	ch := make(chan board.Tick)
-	err := b.StreamTicks(e.cancelCtx, interrupts, ch, nil)
+	err := b.StreamTicks(e.cancelCtx, []board.DigitalInterrupt{e.A, e.B}, ch, nil)
```

Comment:
could you also update the tests for this file to not use the di.Tick interface - I think you just have to delete the tests that check if calling the Tick doesn't error.

Reply:
this isn't an easy change, the Tick function needs to be called to update the position of the encoder in the tests. I think we will need a Tick helper function for tests after we remove from the interface. Can do this change in another PR if thats okay

Reply:
I'd directly handle that by having stream ticks update the Tick value in the injected board if you get to it. Or a shared variable that you increment in the injected board.

Simple and contained is better.